hatch/src/cli/commands/new.rs

```diff
@@ -29,74 +44,57 @@ impl<'command> Command<'command> for New {
     SubCommand::with_name(&self.name)
       .about("Creates a new project. (default = shared library)")
 
-      .arg(Arg::with_name("PROJECT_NAME")
+      .arg(Arg::with_name("PROJECT_NAMES")
            .help("Name of project")
-           .required(true).takes_value(true))
+           .takes_value(true).max_values(1)
+           .required(true))
 
       .arg(Arg::with_name("bin")
            .help("Generate a stand alone executable")
-           .long("bin").short("b").required(false)) 
+           .long("bin").short("b")
+           .required(false)) 
 
       .arg(Arg::with_name("static")
            .help("Generate a static library")
-           .long("static").short("s").conflicts_with("bin").required(false))
+           .long("static").short("s").conflicts_with("bin")
+           .required(false))
 
       .arg(Arg::with_name("VERSION")
            .help("Set the project version")
-           .long("version").short("v").required(false).takes_value(true))
+           .long("version").short("v").takes_value(true)
+           .required(false))
   }
 
   fn subcommand_name(&self) -> &'static str {
     self.name
   }
 
-  fn execute(&self, args: &ArgMatches<'command>) -> Result<Project, HatchError> {
-    match fs::create_dir(self.project_path(args) + self.project_name(args).unwrap().as_str()) {
-      Err(e) => Err(HatchError::from(e)),
+  fn execute(&self, args: &ArgMatches<'command>) -> Vec<HatchResult<Project>> {
+    let name = self.project_names(args).into_iter().nth(0).unwrap();
+
+    let dir_path = self.project_path(args) + &name[..];
+    let hatch_file = self.project_path(args) + &name[..] + "/Hatch.yml";
+
+    match fs::create_dir(&dir_path) {
+      Err(e) => vec![Err(HatchError::from(e))],
       Ok(_) => {
+        let version = self.project_version(args);
+        let kind = self.project_kind(args);
+
         let mut yaml_output = String::new();
-        
-        write!(&mut yaml_output, "name: {}\nversion: {}\nbuild: {}\n",
-               self.project_name(args).unwrap(),
-               self.project_version(args),
-               self.project_kind(args));
-
-        match fs::File::create(self.project_path(args)
-                               + self.project_name(args).unwrap().as_str()
-                               + "/Hatch.yml") {
-          Err(e) => Err(HatchError::from(e)),
+
+        write!(&mut yaml_output, "name: {}\nversion: {}\nbuild: {}\n", &name, &version, &kind);
```

Comment:
Do we not like building a Yaml object and using the Emitter system?  Just curious, I am not opposed to do it outselves, but letting them generate the Yaml might be easier.

Reply:
They dont generate anything with yaml_emitter.  Its basically is a fancy writer that I think verifies the yaml.  As for right now I decided to leave it as just valid but raw yaml text.

Reply:
Fair enough, it doesn't hurt that all of our Yaml is probably going to be pretty simple.